StackTracerCore/Debugger.cs

```diff
@@ -46,175 +47,168 @@ private static void TraceWarning(string message, Exception ex)
+                            stats.ThreadCount = threads.Count();
 
-                        var timeThreadStart = DateTime.UtcNow;
-                        var threads = runtime.Threads.Where(t => t.IsAlive);
-                        stats.TimeFetchingThreads = DateTime.UtcNow.Subtract(timeThreadStart).TotalMilliseconds;
-                        stats.ThreadCount = threads.Count();
-
-                        foreach (ClrThread thread in threads.Take(MaxThreadsToDump))
-                        {
-                            ManagedThread t = new ManagedThread
-                            {
-                                OSThreadId = thread.OSThreadId,
-                                GcMode = thread.GCMode,
-                                IsBackground = thread.IsBackground,
-                                IsFinalizer = thread.IsFinalizer,
-                                IsGCSuspendPending = thread.IsGCSuspendPending,
-                                LockCount = thread.LockCount,
-                                ManagedThreadId = thread.ManagedThreadId,
-                                IsUserSuspended = thread.IsUserSuspended
-                            };
-
-                            var stackTrace = new List<string>();
-
-                            DateTime dtStackTrace = DateTime.UtcNow;
-                            var threadStackTrace = thread.EnumerateStackTrace();
-                            stats.TimeFetchingStackFrames += DateTime.UtcNow.Subtract(dtStackTrace).TotalMilliseconds;
-
-                            foreach (ClrStackFrame frame in threadStackTrace)
+                            foreach (ClrThread thread in threads.Take(MaxThreadsToDump))
                             {
-                                if(frame == null)
+                                ManagedThread t = new ManagedThread
                                 {
-                                    continue;
-                                }
-
-                                if (frame.Kind == ClrStackFrameKind.ManagedMethod)
+                                    OSThreadId = thread.OSThreadId,
+                                    GcMode = thread.GCMode,
+                                    IsBackground = thread.IsBackground,
+                                    IsFinalizer = thread.IsFinalizer,
+                                    IsGCSuspendPending = thread.IsGCSuspendPending,
+                                    LockCount = thread.LockCount,
+                                    ManagedThreadId = thread.ManagedThreadId,
+                                    IsUserSuspended = thread.IsUserSuspended
+                                };
+
+                                var stackTrace = new List<string>();
+
+                                var threadStackTrace = thread.EnumerateStackTrace();
+                                TraceLine($"EnumerateStackTrace returned = {threadStackTrace.Count()} frames", false);
```

Comment:
Calling `.Count()` here will enumerate the stack trace twice, which can be slow.  I suggest moving this line after the foreach statement and just keeping track of a count as you go.  e.g.:

```C#
int frameCount = 0;
foreach (ClrStackFrame frame in threadStackTrace)
{
    frameCount++;

    ... // All of the code you currently have in the foreach loop
}
TraceLine($"EnumerateStackTrace returned = {frameCount} frames", false);
```

Reply:
done